### DISCUSSION TITLE
I want to add Giscus to the end of each blog post

### DISCUSSION THREAD

**Discussion by MisterMunchkin** [facebook/docusaurus#8140] [General]
# Add Giscus or other comment implementations at the bottom of each blog post

There's currently no way of adding components to the end of each blog post like Giscus.

## Infos

- Site source code: https://github.com/MisterMunchkin/blog-robin-dalmy
- Site URL: https://blog.robindalmy.com/
- Feature URL: https://blog.robindalmy.com/ (Still in development)

## Screenshots

<img width="1412" alt="Screen Shot 2022-09-28 at 1 33 54 PM" src="https://user-images.githubusercontent.com/9417970/192695946-9ab6aafe-ae9a-487c-b75b-dcb1c91ad06d.png">

## Implementation

I swizzled BlogPostItem/Footer just so I can add Giscus on every blog post. I would also need to set up some conditions to make sure Giscus doesn't pop up on the recent post lists on the home page.

**Comment by slorber** [comment]
Hey, not sure if this is a question or what you expect us to answer here.

You can add a comment system to Docusaurus, see for example https://madrus4u.com/blog/blog-comments

`BlogPostItem/Footer` access a `isBlogPostPage` boolean that you can use to avoid displaying comments in paginated lists. You can also user the ` const {isBlogPostPage} = useBlogPost();` in a footer wrapper in case you don't want to eject

Edit: see also https://github.com/facebook/docusaurus/pull/8088/files

**Reply by MisterMunchkin** [reply]
Hi, I was following the instructions from this discussion #5468. Apologies if I didn't do this correctly, I'm new to discussions on GitHub 😅.

I just decided to write this discussion because there was a link to the #5468 discussion thread after swizzling `BlogPostItem/Footer` because it was unsafe.

**Reply by slorber** [reply]
Ah I see.

I clarified the instructions because it was not expected to create a brand new github discussion

> Report each use case as a separate discussion thread **(in this discussion)**

But I guess it's fine and maybe we'll want to have one unique discussion per customization use-case in the future (less messy), we probably need to change the process a bit :)

I guess we'll want to create a "discussion category" of type "site customizations feedback" or something 🤷‍♂️

**Reply by MisterMunchkin** [reply]
Ahh, alright. Yeah, I think separate discussions for each would be nice too. But for now, should I transfer this over to #5468?

**Reply by slorber** [reply]
no it's ok, I noted to create a new process and update the issue above
